bibliopixel/remote/control.py

```diff
@@ -122,16 +122,18 @@ def __init__(self, layout, animations,
         self.triggers = {}
         for trigger in triggers:
             typename = trigger.pop('typename')
-            if typename:
+            events = trigger.pop('events')
+            if typename and events:
                 importer.import_symbol(typename)  # attempt early to fail early
-                self.triggers.setdefault(typename, []).append(trigger)
+                self.triggers.setdefault(typename, []).extend((events, trigger))
             else:
-                raise ValueError('Triggers require a `typename` field!')
+                raise ValueError('Triggers require `typename` and `events` fields!')
 
-        for typename, configs in self.triggers.items():
+        for typename, trigger in self.triggers.items():
```

Comment:
Clearer and shorter is:

    for typename, (events, kwargs) in self.triggers.items():